exonum/src/encoding/segments.rs

```diff
@@ -197,6 +197,7 @@ impl<'a> SegmentField<'a> for RawMessage {
     }
 }
 
+#[cfg_attr(feature = "cargo-clippy", allow(use_self))]
```

Comment:
Why do we need it here?

Reply:
Because of:
```
error: unnecessary structure name repetition
   --> exonum\src\encoding\segments.rs:228:46
    |
228 |     fn extend_buffer(&self, mut buffer: &mut Vec<u8>) {
    |                                              ^^^^^^^ help: use the applicable keyword: `Self`
    |
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.211/index.html#use_self
```

Reply:
One more reason to disable this lint, as for me.

Reply:
I will remove use_self allowing and in another places too.

Reply:
I agree with popzxc, and will disable this one. But I propose to enable it time to time, and verify all sources to catch and fix exactly indisputable places.